subsys/bluetooth/host/keys.c

```diff
@@ -62,17 +96,28 @@ struct bt_keys *bt_keys_get_addr(uint8_t id, const bt_addr_le_t *addr)
 
 #if IS_ENABLED(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
 	if (first_free_slot == ARRAY_SIZE(key_pool)) {
-		struct bt_keys *oldest = &key_pool[0];
+		struct bt_keys *oldest = NULL;
 		bt_addr_le_t oldest_addr;
 
-		for (i = 1; i < ARRAY_SIZE(key_pool); i++) {
+		for (i = 0; i < ARRAY_SIZE(key_pool); i++) {
 			struct bt_keys *current = &key_pool[i];
+			bool key_in_use = (CONFIG_BT_MAX_CONN > 1) && key_is_in_use(i);
```

Comment:
Hmm, if `CONFIG_BT_MAX_CONN == 1` then no keys can be in use. 

It's a bit difficult for me to fully understand if there are any side effect here, because I guess that if no keys are in use, then we possibly overwrite the key for the current connection. Is that safe to do?

Reply:
The logic is if `CONFIG_BT_MAX_CONN == 1` then the bonding device is the only connected device which means all available key storage slots (1 or more) are considered not in use (by a connected device).  Note if there was a key slot previously used by the bonding device that would have been detected before the code above runs.  See:
https://github.com/zephyrproject-rtos/zephyr/blob/21ac7efc5c2629ca74d657bf52b0ed0a4ff62209/subsys/bluetooth/host/keys.c#L87-L89

Reply:
Alright :)